### PR TITLE
Feature/cookies

### LINKED ISSUE #45: Persisting Data
Implement Local Storage or other possible solution for data persisting without a user

### PR DESCRIPTION
**Pull request by alivaditis**
This PR will close #45 

What I did:
COOKIES
added cookies for persisting data
added a line to clear cookies in the fill cart command

Did this break anything?
- [x] No
- [ ] Yes

Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ] Super small fix (Corrected a typo, removed a comment, etc.)
- [ ] Skip all the other stuff and briefly explain the fix.

Checklist:
- [x] If this code needs to be tested, all tests are passing
- [x] The code runs locally
- [x] There are comments where clarification/ organization is needed
- [x] The code is DRY. If it's not, I tried my best
- [x] I reviewed my code before pushing

What's next?
finishing touches

### CODE REVIEW

File: cypress/e2e/checkout_spec.cy.js [hunk 1]
```diff
@@ -1,74 +1,77 @@
...
-//       statusCode: 500,
-//       fixture: 'submitFail.json'
-//     }).as('createOrderForm')
-//       .get('.checkout__form__submit').click()
-//       .wait('@createOrderForm')
-//       .get('.checkout__form__fail').contains('p','Your order request could not be processed at this time. Please try again later.')
-//     })
-// })
\ No newline at end of file
+describe('checkout', () => {
```
Comment (rachelsoae) [inline cypress/e2e/checkout_spec.cy.js:1]: All tests are passing. Great work.

File: src/components/App.js [hunk 2]
```diff
@@ -15,13 +16,25 @@ import ProductDetail from './ProductDetail/ProductDetail';
...
+  const [cookies, setCookie] = useCookies(['shoppingBag']);
+  const [shoppingBag, setShoppingBag] = useState([])
+    
+  const { loading, error, data } = useQuery(GET_ALL_ITEMS)
   const [totalPrice, setTotalPrice] = useState(0);
   const [items, setItems] = useState([]);
   const [itemsForDisplay, setItemsForDisplay] = useState([]);
   const [successMessage, setSuccessMessage] = useState('');
 
+  useEffect(() => {
```
Comment (rachelsoae) [inline src/components/App.js:28]: Very cool! Thanks for handling this sick feature!

**Review by rachelsoae — APPROVED** [review]

File: src/components/App.js (hunk 2 above)
Comment (taranveersingh93) [reply src/components/App.js:28]: +1

**Review by taranveersingh93 — CHANGES_REQUESTED** [review]
The tests pass. The feature works locally. Would love a test comparing the length of shopping cart before and after a refresh button click?

File: cypress/e2e/checkout_spec.cy.js [hunk 3]
```diff
@@ -3,6 +3,7 @@ describe('checkout', () => {
     cy.stubRequestsDynamically()
     cy.visit("http://localhost:3000/#products")
     cy.wait("@GetAllItems")
+    cy.clearCookies()
```
Comment (taranveersingh93) [inline cypress/e2e/checkout_spec.cy.js:6]: nice

File: cypress/e2e/checkout_spec.cy.js [hunk 4]
```diff
@@ -74,4 +75,15 @@ describe('checkout', () => {
...
+    cy.reload()
+      .get('.checkout__item').should('have.length', '3')
+      .get('.checkout__item').first().contains('b', '2x Bracelet')
+      .get('.checkout__item').first().contains('p', 'Color: moss')
+      .get('.checkout__item').first().contains('p', 'Size: M')
+      .get('.checkout__item').last().contains('1x Dog Leash')
+      .get('.checkout__item').last().contains('Color: lime')
+      .get('.checkout__pricing').contains('Estimated Total')
+      .get('.checkout__pricing').contains('$130.00')
+  })
```
Comment (taranveersingh93) [inline cypress/e2e/checkout_spec.cy.js:88]: Beautiful and thorough

File: cypress/e2e/checkout_spec.cy.js [hunk 5]
```diff
@@ -74,4 +75,15 @@ describe('checkout', () => {
       .wait('@createOrderForm')
       .get('.checkout__form__fail').contains('p','Your order request could not be processed at this time. Please try again later.')
     })
+  it('the user should have same items in cart on refresh', () => {
+    cy.reload()
```
Comment (taranveersingh93) [inline cypress/e2e/checkout_spec.cy.js:79]: The network request is stubbed but if it doesn't break, could you add `cy.wait("@GetAllItems")` here?

**Review by taranveersingh93 — CHANGES_REQUESTED** [review]
Just one comment if you could clarify?

**Review by taranveersingh93 — APPROVED** [review]
Thank you for this functionality

**Pull request merged**